pkg/provider/azure_local_services.go

```diff
@@ -544,3 +543,84 @@ func (az *Cloud) cleanupLocalServiceBackendPool(
 	}
 	return lbs, nil
 }
+
+// checkAndApplyLocalServiceBackendPoolUpdates if the IPs in the backend pool are aligned
+// with the corresponding endpointslice, and update the backend pool if necessary.
+func (az *Cloud) checkAndApplyLocalServiceBackendPoolUpdates(lb network.LoadBalancer, service *v1.Service) error {
+	serviceName := getServiceName(service)
+	endpointsNodeNames, err := az.getLocalServiceEndpointsNodeNames(service)
+	if err != nil {
+		return err
+	}
+	var expectedIPs []string
+	for nodeName := range endpointsNodeNames {
+		ips := az.nodePrivateIPs[strings.ToLower(nodeName)]
+		for ip := range ips {
+			expectedIPs = append(expectedIPs, ip)
+		}
+	}
+	currentIPsInBackendPools := make(map[string][]string)
+	for _, bp := range *lb.BackendAddressPools {
+		bpName := pointer.StringDeref(bp.Name, "")
+		if localServiceOwnsBackendPool(serviceName, bpName) {
+			var currentIPs []string
+			for _, address := range *bp.LoadBalancerBackendAddresses {
+				currentIPs = append(currentIPs, *address.IPAddress)
+			}
+			currentIPsInBackendPools[bpName] = currentIPs
+		}
+	}
+	az.applyIPChangesAmongLocalServiceBackendPoolsByIPFamily(*lb.Name, serviceName, currentIPsInBackendPools, expectedIPs)
```

Comment:
how would the error be returned back on failures here?

Reply:
It will be in the service events.